pisces/io/response.py

```diff
@@ -697,3 +703,878 @@ def read_pazfir(path, input_samp_rate, calib=None, calper=None, input_units='NM'
+        specified by the KBCore Schema.  Chanid and Inid are automatically assigned based 
+        on channel and response indices in the object
+    instrumentDict: dict
+        Dictionary containing all of the sensor metadata in Inventory object as 
+        specified by the KBCore Schema. Inid is automatically assigned based on the response
+        index within the object and will correspond to the correct chanid in the sensor table
+
+    Examples:
+    ---------
+    sensor, instrument  = sxml2pazfir('stationxml_path', write_tables = False)
+    sxml2pazfir(InventoryObject, out_freq = 1.0, dir_path = 'path_to_directory')
+
+    """
+
+    # input can be inv object or file path as string
+    if type(input_xml) is str:
+        invObj = read_inventory(input_xml)
+    else:
+        invObj = input_xml
+
+    if dir_path is None:
+        dir_path = os.getcwd()
+    
+    inid = 0
+    chanid = 0
+
+    # sta, chan, time, endtime, inid, chanid, jdate, calratio, calper, thsift (clock errors), instant
+    sensorDict = {'tablename':'sensor', 'sta':[], 'chan':[], 'time':[], 'endtime':[], 'inid':[], 'chanid':[], \
+                  'jdate':[], 'calratio':[], 'calper':[], 'tshift':[], 'instant':[]}
+     # inid, insname, instype, band, digital (d/a), samprate, ncalib, ncalper, dir, dfile, rsptype
+    instrumentDict = {'tablename':'instrument','inid':[], 'insname':[], 'instype':[], 'band':[], 'digital':[], \
+                     'samprate':[], 'ncalib':[], 'ncalper':[], 'dir':[], 'dfile':[], 'rsptype':[]} 
+
+    for net in invObj.networks:
+        netCode = net.code
+        for sta in net.stations:
+            staCode = sta.code
+            for chan in sta.channels:
+                chanCode = chan.code
+                locCode =chan.location_code
+                sampRate = chan.sample_rate
+                sensorDesc = chan.sensor.description
+                startDate = chan.start_date
+                endDate = chan.end_date
+                dataLogger = chan.data_logger
+                preAmp = chan.pre_amplifier
+                
+                # for table formation
+                chanLocCode = chanCode + locCode
+                bandCode = chanCode[0].lower()
+                jDate = int(startDate.strftime('%Y%j'))
+
+                if sensorDesc is None:
+                    sensorDesc = '-'
+
+                if startDate is None:
+                    startDate = UTCDateTime(-9999999999.999)
+                    jDate 
+                if endDate is None:
+                    endDate =  UTCDateTime(9999999999.999)
+                
+                #check if response information exists, if so, go into response loop
+                if len(chan.response.response_stages) > 0: 
+                    sensitivity, out_freq, fileName =  write_pazfir(chan.response, staCode, chanCode, startDate, out_freq = out_freq, dir_path = dir_path, network=netCode, location=locCode, \
```

Comment:
Could you write `write_pazfir` to accept 0 response stages (doesn't write anything), so that this check isn't necessary?

Reply:
Well...if there aren't any response stages all that would be output is a file only containing a header or I'd have to error out that there's no input to write...

Reply:
👍